### DISCUSSION TITLE
QQL support

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#225] [Q&A]
Hi team,
Actually we are trying to query timebase as per below subqueries but getting unknown identifier Column 1, 
can you please us with same.

Select column1 from ( select * from stream1) 
....this is working fine

Select column1 from (select * from stream1 where column2 =='abc' union select * from stream1 where column2=='xyz')
....But when i try this it is giving error that UnKnownIdentifierException: unknown identifier :COLUMN1

If in above query I do "select symbol from(...union....)" it is working as expected.

Please help me know why Timebase wont selecting custom columns and how it will work.

FYI
I have created stream stream1 using QQL as per below:
CREATE DURABLE STREAM "stream1"(
      CLASS "A.B.C.MessageClass" 'Stream test' (
          "column1" 'column1' VARCHAR
          "column2" 'column 2' VARCHAR
  );
)
OPTIONS (FIXEDTYPE; PERIODICITY ='IRREGULAR'; HIGHAVAILIBILTY =TRUE)

I am using timebase shell for quering data

**Comment by ankitsoin** [comment]
I have already upgraded to 6.1 version

**Reply by alex-karpovich** [reply]
```Select column1 from (select * from stream1 where column2 =='abc' union select * from stream1 where column2=='xyz')```

such statements ````SELECT FROM (SELECT ...)     ```` is not yet implemented 

anyway please clarify your real use case, I believe it can be done using regular SELECTs

**Reply by ankitsoin** [reply]
I have a bit complicated logic so will try and will elaborate to you on soon.

**Reply by ShivaniSK** [reply]
Hello @alex-karpovich ,
Is the SELECT FROM (SELECT ...)    sub- queries supported in enterprise edition?

**Comment by ankitsoin** [comment]
While quering do we have any where condition on spaces?
To make my query easy in timebase shell.

**Reply by alex-karpovich** [reply]
spaces is not present in QQL, in can be defined on SelectionOptions

**Comment by ankitsoin** [comment]


**Reply by alex-karpovich** [reply]
select Time, symbol, column1, column2/1000000.0 from "stream" 

please read https://kb.timebase.info/community/development/qql/QQL%205.5/qql-tut-functions to find additional details
